Modules/mathmodule.c

```diff
@@ -3017,103 +3017,105 @@ Raises ValueError if the arguments are negative or if k > n.
 
 static PyObject *
 math_comb_impl(PyObject *module, PyObject *n, PyObject *k)
-/*[clinic end generated code: output=bd2cec8d854f3493 input=565f340f98efb5b5]*/
+/*[clinic end generated code: output=bd2cec8d854f3493 input=b2160da8fe59df60]*/
 {
-    PyObject *val = NULL,
-        *temp_obj1 = NULL,
-        *temp_obj2 = NULL,
-        *dump_var = NULL;
+    PyObject *result = NULL, *factor = NULL, *temp;
     int overflow, cmp;
-    long long i, terms;
+    long long i, factors;
+
+    n = PyNumber_Index(n);
+    if (n == NULL)
+        return NULL;
+    k = PyNumber_Index(k);
+    if (k == NULL) {
+        Py_DECREF(n);
+        return NULL;
+    }
 
     cmp = PyObject_RichCompareBool(n, k, Py_LT);
     if (cmp < 0) {
-        goto fail_comb;
+        goto error;
     }
     else if (cmp > 0) {
-        PyErr_Format(PyExc_ValueError,
-                     "n must be an integer greater than or equal to k");
-        goto fail_comb;
+        PyErr_SetString(PyExc_ValueError,
+                        "n must be an integer greater than or equal to k");
+        goto error;
     }
 
-    /* b = min(b, a - b) */
-    dump_var = PyNumber_Subtract(n, k);
-    if (dump_var == NULL) {
-        goto fail_comb;
-    }
-    cmp = PyObject_RichCompareBool(k, dump_var, Py_GT);
-    if (cmp < 0) {
-        goto fail_comb;
+    /* k = min(k, n - k) */
+    temp = PyNumber_Subtract(n, k);
+    if (temp == NULL) {
+        goto error;
     }
-    else if (cmp > 0) {
-        k = dump_var;
-        dump_var = NULL;
+    cmp = PyObject_RichCompareBool(k, temp, Py_GT);
+    if (cmp > 0) {
+        Py_SETREF(k, temp);
     }
     else {
-        Py_DECREF(dump_var);
-        dump_var = NULL;
+        Py_DECREF(temp);
+        if (cmp < 0 && PyErr_Occurred()) {
```

Comment:
Do we need the `PyErr_Occurred()` check here?

Reply:
> We definitely needed it in the old code

Why? There's no code path in `PyObject_RichCompareBool` that leads to `-1` being returned without an error set. We don't check for `PyErr_Occurred()` after every `== NULL` check, so why would we do it here? It's completely redundant code, and it should be removed. What sort of future changes could possibly necessitate this level of defensive coding?

Reply:
It was my mistake. I missed `PyObject_RichCompareBool` with `PyUnicode_Compare` or something like.

Reply:
Yes, apologies. I think our conversations crossed and I missed your other comment. And my comment in this thread was borderline rude. Sorry.

Reply:
It did not look rude to me. No apologies needed. Thank you and @pablogsal for your review!